models/method.go

```diff
@@ -0,0 +1,99 @@
+package models
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/iancoleman/strcase"
+	"github.com/lightninglabs/lightning-api-ng/defs"
+	"github.com/lightninglabs/lightning-api-ng/markdown"
+)
+
+type Method struct {
+	Service *Service
+
+	Name               string
+	Description        string
+	Source             string
+	CommandLine        string
+	CommandLineHelp    string
+	RequestType        string
+	RequestFullType    string
+	RequestTypeSource  string
+	RequestStreaming   bool
+	ResponseType       string
+	ResponseFullType   string
+	ResponseTypeSource string
+	ResponseStreaming  bool
+}
+
+// NewMethod creates a new method from a method definition
+func NewMethod(methodDef *defs.ServiceMethod, service *Service) *Method {
+	m := &Method{}
+	m.Service = service
+	m.Name = methodDef.Name
+	m.Description = parseDescription(methodDef.Description)
+	m.Source = methodDef.Source
+	m.CommandLine = methodDef.CommandLine
+	m.CommandLineHelp = methodDef.CommandLineHelp
+	m.RequestType = methodDef.RequestType
+	m.RequestFullType = methodDef.RequestFullType
+	m.RequestTypeSource = methodDef.RequestTypeSource
+	m.RequestStreaming = methodDef.RequestStreaming
+	m.ResponseType = methodDef.ResponseType
+	m.ResponseFullType = methodDef.ResponseFullType
+	m.ResponseTypeSource = methodDef.ResponseTypeSource
+	m.ResponseStreaming = methodDef.ResponseStreaming
+	return m
+}
+
+// IsDeprecated returns true if the method contains the word "deprecated" in
+// the description
+func (m *Method) IsDeprecated() bool {
+	return strings.Contains(strings.ToLower((m.Description)), "deprecated")
+}
+
+func (m *Method) StreamingDirection() string {
+	if m.RequestStreaming && m.ResponseStreaming {
```

Comment:
Golang has the (IMO) great feature of allowing `switch` statements with just boolean cases. So this could be re-written to the more readable (IMO) version:
```
switch {
case m.RequestStreaming && m.ResponseStreaming:
...
case m.RequestStreaming:
...
case m.ResponseStreaming:
...
default:
    return ""
}
```

Reply:
I updated all if/else statements to use `switch` when it appeared to make sense to me. Let me know if you find any left.